backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramRefreshManager.java

```diff
@@ -96,16 +97,20 @@ public void initialize(UUID projectId, DiagramCreationParameters diagramCreation
     public void refresh(UUID projectId, DiagramCreationParameters diagramCreationParameters) {
         long start = System.currentTimeMillis();
 
-        this.computeDiagram(projectId, diagramCreationParameters);
+        this.computeDiagram(projectId, diagramCreationParameters, Optional.of(this.diagram));
 
         long end = System.currentTimeMillis();
         this.timer.record(end - start, TimeUnit.MILLISECONDS);
 
         this.sink.next(new DiagramRefreshedEventPayload(this.diagram));
     }
 
-    private void computeDiagram(UUID projectId, DiagramCreationParameters diagramCreationParameters) {
-        Diagram unlayoutedDiagram = this.diagramService.create(diagramCreationParameters);
+    private void computeDiagram(UUID projectId, DiagramCreationParameters diagramCreationParameters, Optional<Diagram> optionalPrevDiagram) {
+        // @formatter:off
+        Diagram unlayoutedDiagram = optionalPrevDiagram
+                .map(prevDiagram -> this.diagramService.refresh(diagramCreationParameters, this))
+                .orElse(this.diagramService.create(diagramCreationParameters));
+        // @formatter:on
```

Comment:
I don't understand what you are trying to achieve in this commit since the commit message only talks about a technical goal (propagate the diagram manager). Why is the DiagramManager using the DiagramService which uses the DiagramManager? That's odd.